plugins/source_docs.go

```diff
@@ -14,20 +15,100 @@ import (
 //go:embed templates/*.go.tpl
 var templatesFS embed.FS
 
+type SourceDocsFormat string
+
+const (
+	SourceDocsFormatMarkdown = "markdown"
+	SourceDocsFormatJSON     = "json"
+)
+
+var SourceDocsFormats = []SourceDocsFormat{
+	SourceDocsFormatMarkdown,
+	SourceDocsFormatJSON,
+}
+
+func (s SourceDocsFormat) String() string {
+	return string(s)
+}
+
+func (s SourceDocsFormat) Validate() error {
+	for _, f := range SourceDocsFormats {
+		if s == f {
+			return nil
+		}
+	}
+	return fmt.Errorf("invalid format: %v", s.String())
+}
+
 // GenerateSourcePluginDocs creates table documentation for the source plugin based on its list of tables
-func (p *SourcePlugin) GenerateSourcePluginDocs(dir string) error {
+func (p *SourcePlugin) GenerateSourcePluginDocs(dir string, format SourceDocsFormat) error {
 	if err := os.MkdirAll(dir, os.ModePerm); err != nil {
 		return err
 	}
 
 	SetDestinationManagedCqColumns(p.Tables())
 
+	switch format {
+	case SourceDocsFormatMarkdown:
+		return p.renderTablesAsMarkdown(dir)
+	case SourceDocsFormatJSON:
+		return p.renderTablesAsJSON(dir)
+	default:
+		return fmt.Errorf("unsupported format: %v", format.String())
+	}
+}
+
+type jsonTable struct {
```

Comment:
I think this is fine but can't we re-use the same tables definition that we already have? we also encode them to json or we want a different structure.

Reply:
I tried this initially, and if I recall correctly the issue I faced was that column type came out as an integer, which isn't very useful without the mappings. So I figured that it would be good to decouple the GRPC representation from the one we use here, but I could be convinced that we shouldn't do that, and figure out a way to turn the type to a string here instead?

Reply:
Yeah I don't have a good idea right now so maybe we can merge this as is and maybe fix later on if it becomes a maintains burden. But actually now that I think about it why the gRPC representation is not good ? especially if it's intended to be read by computers? Maybe we can have some flag that will marshal the `Type` in a different way.

Reply:
I think one use case for this is going to be outputting the schema in a different format than the markdown we support. Maybe a different markdown format, maybe HTML? But having a number for the type is not very useful then, you'd either need access to the plugin-sdk or copy-paste the list of constants to know how it maps to a string. I also think it's probably good if we can evolve the internal representations separately. We may even move away from JSON for internal communication purposes if it turns out to be a drag on performance, but we'd still want to keep this.